src/shadowbox/server/shared_metrics.spec.ts

```diff
@@ -21,25 +23,44 @@ import {AccessKeyConfigJson} from './server_access_key';
 import {ServerConfigJson} from './server_config';
 import {
   CountryUsage,
-  DailyFeatureMetricsReportJson,
-  HourlyServerMetricsReportJson,
   MetricsCollectorClient,
   OutlineSharedMetricsPublisher,
+  SharedMetricsPublisher,
   UsageMetrics,
 } from './shared_metrics';
 
 describe('OutlineSharedMetricsPublisher', () => {
+  let clock: ManualClock;
```

Comment:
The shared state prevents tests from running in parallel. Is this helping much?

I'd much rather see a set up call before each test, since that is more readable and makes the dependency clear. The hidden dependency hurts readability.

Reply:
I think all that kind of magic makes the tests harder to understand. It's also not clear what is actually needed by the test.

The "Reads from config" is particularly mind-boggling as it has no action in it!

It's clearer if the test is self-contained, with no access to external state.
You can use helper methods if needed, and still use Jasmine's describe functionality. For example:

```ts
  describe('Enable/Disable', () => {
    it('Mirrors config', () => {
      const serverConfig = new InMemoryConfig({serverId: 'server-id'});
      const publisher = makeTestPublisher({serverConfig})

      expect(publisher.isSharingEnabled()).toBeFalse();

      publisher.startSharing();
      expect(publisher.isSharingEnabled()).toBeTrue();
      expect(serverConfig.mostRecentWrite.metricsEnabled).toBeTrue();

      publisher.stopSharing();
      expect(publisher.isSharingEnabled()).toBeFalse();
      expect(serverConfig.mostRecentWrite.metricsEnabled).toBeFalse();
    });
```

Note how makeTestPublisher allows for overriding fields.

It adds two lines, but all the logic is in one place. I don't need to look somewhere else to understand this test. And I don't need to worry about parallel execution.